docs/intro/from-ground-up.rst

```diff
@@ -498,151 +527,160 @@ to
+In addition to signature inspection, it also handles
+typing.Optional and typing.Union, and allows to create a build plan
+for dependency trees, indirect dependencies: that's allowed to annotate
+an argument as another :class:`~.Injectable` subclass.
 
-1. A list of possible inputs for the page objects. This helps with
+``web-poet`` is not using andi_ on its own; ``web-poet``'s role
+is to standardize things + provide some helpers to write the
+extraction code easier:
+
+1. Standardize a list of possible inputs for the page objects. This helps with
    reusability of extraction code across different environments. For example,
    if you want to support extraction from raw HTTP response bodies, you
    need to figure out how to populate :class:`~.ResponseData` in the
    given environment, and that's all.
 
-   Users are free to define their own inputs, but they may be less portable
-   across environments.
+   Users are free to define their own inputs (input types), but they
+   may be less portable across environments - which can be fine.
 
    Currently only :class:`~.ResponseData` is defined in web-poet.
 
-2. Interface for the Page Object itself. This allows to have a code which can
-   instantiate and use a Page Object without knowing about its
-   implementation upfront. ``web-poet`` requires you to use a base class,
-   and defines the semantics of ``to_item()`` method.
+2. Define an interface for the Page Object itself. This allows to
+   have a code which can instantiate and use a Page Object without knowing
+   about its implementation upfront. ``web-poet`` requires you to
+   use a base class (:class:`~.Injectable`), and defines the
+   semantics of ``to_item()`` method.
+
+
+Then, framework's role is to:
+
+1. Figure out which inputs a Page Object needs, likely using andi_ library.
+2. Create all the necessary inputs. For example, creating
+   :class:`~.ResponseData` instance may involve making an HTTP request.
+3. Create a Page Object instance, passing it the inputs it needs.
+4. Depending on a task, either return a newly created Page Object
+   instance to the user, or call some predefined method
+   (a common case is ``to_item``).
+
+For example, web-poet + Scrapy integration package (scrapy-poet_)
+may inspect a WebPage subclass you defined, figure out it needs
+:class:`~.ResponseData` and nothing else, fetch scrapy's TextResponse,
+create ``ResponseData`` instance from it, create your
+Page Object instance, and pass it to a spider callback.
+
+Finally, the Developer's role is to:
+
+1. Write a Page Object class, likely website-specific, following ``web-poet``
+   standards. The extraction code should define the inputs it needs
+   (such as "body of HTTP response", "Chrome DOM tree snapshot",
+   "crawl state"); it shouldn't fetch these inputs itself.
+2. Pass the Page Object class to a framework, in a way defined by the
+   framework.
+3. Depending on a task, either get a Page Object instance, or get the data
```

Comment:
I don't understand this phrase.

Reply:
Is it better now?

Reply:
Much better, now I understand it. After reading it I'm ready to use web-poet :-) I left some minor suggestions